### PR TITLE
Water Sampling Card

### PR DESCRIPTION
**Pull request by tleraj**
The purpose of this PR is to close https://github.com/aqualinkorg/aqualink-app/issues/644.
Things to do:
 - [x] Handle design edge case for long survey point names at the card bottom
 - [x] Implement `VIEW UPLOAD` button functionality

### Design
Screenshot_20211224_173154

### CODE REVIEW

File: packages/website/src/common/SiteDetails/index.tsx [hunk 1]
```diff
@@ -37,6 +36,7 @@ import {
 } from "../../store/Sites/selectedSiteSlice";
 import LoadingCard from "./LoadingCard";
 import WaterSamplingCard from "./WaterSampling";
+import { hasSitesSondeData } from "../../store/Sites/helpers";
```
Comment (ericboucher) [inline packages/website/src/common/SiteDetails/index.tsx:39]: Maybe call this functio `sondeDataCheck` or `siteHasSondeData`?

**Review by ericboucher — COMMENTED** [review]
Looks great

Minor name change for a function.
And if it's easy, let's
- [ ] display the wind_speed and significant wave height for sites that have a spotter now that we are fetching them.

**Review by ericboucher — APPROVED** [review]

File: packages/website/src/common/Chart/ChartWithTooltip.tsx [hunk 2]
```diff
@@ -72,105 +72,108 @@ function ChartWithTooltip({
...
+        tooltipModel.caretY -
+        ((surveyId ? 30 : 0) + nValues * 20 + 48);
+
+      if (
+        nValues > 0 &&
+        moment(date).isBetween(
+          moment(startDate || last(filteredDailyData)?.date),
+          moment(endDate || filteredDailyData?.[0]?.date),
+          undefined,
+          "[]"
```
Comment (ericboucher) [inline packages/website/src/common/Chart/ChartWithTooltip.tsx:157]: Maybe add a few comments around here to make it easier to discover for others or ourselves in a few month

File: packages/website/src/common/Chip/index.tsx [hunk 3]
```diff
@@ -60,56 +89,48 @@ const Chip = ({
...
-}
-
 Chip.defaultProps = {
+  live: false,
   href: undefined,
   to: undefined,
   imageText: undefined,
   image: undefined,
   liveText: "LIVE",
-  width: undefined,
```
Comment (ericboucher) [inline packages/website/src/common/Chip/index.tsx:112]: width disappeared from here but is still in the available props, is that normal?

File: packages/website/src/common/SiteDetails/WaterSampling/utils.ts [hunk 4]
```diff
@@ -0,0 +1,20 @@
+import { compact, mapValues, minBy, meanBy } from "lodash";
+import { TimeSeriesData, TimeSeriesRange } from "../../../store/Sites/types";
+
+export const findSondeDataMinAndMaxDates = (ranges?: TimeSeriesRange) => {
+  const minDate = minBy(
+    compact(Object.values(ranges || {}).map((range) => range?.[0]?.minDate)),
+    (date) => new Date(date).getTime()
+  );
+
+  const maxDate = minBy(
```
Comment (ericboucher) [inline packages/website/src/common/SiteDetails/WaterSampling/utils.ts:10]: shouldn't this be maxBy?

File: packages/website/src/constants/sondeConfig.ts [hunk 5]
```diff
@@ -8,7 +8,7 @@ export const sondeConfig: { [key: string]: any | undefined } = {
   },
   turbidity: {
     title: "Turbidity",
-    units: "FNU",
+    units: "fnu",
```
Comment (ericboucher) [inline packages/website/src/constants/sondeConfig.ts:11]: why this change? I think the unit is actually FNU

File: packages/website/src/helpers/map.ts [hunk 6]
```diff
@@ -123,14 +124,17 @@ export const findClosestSurveyPoint = (
...
+        pointName: point.name,
         distance: radDistanceCalculator(
           [siteLng, siteLat],
           getMiddlePoint(polygon)
         ),
       };
     });
 
-  return minBy(distances, "distance")?.pointId;
+  const closestPoint = minBy(distances, "distance");
```
Comment (ericboucher) [inline packages/website/src/helpers/map.ts:135]: if this does not work, return the first one by "survey_point_id"

File: packages/website/src/store/Sites/types.ts [hunk 7]
```diff
@@ -117,37 +117,36 @@ export interface DataRange {
...
+  "odo_concentration",
+  "cholorophyll_concentration",
+  "ph",
+  "salinity",
+  "turbidity",
+] as const;
+
+export type MetricsKeys = typeof metricsKeysList[number];
+
+type CamelCase<S extends string> =
```
Comment (ericboucher) [inline packages/website/src/store/Sites/types.ts:144]: Maybe add a small comment here as well

File: packages/website/src/common/Chip/index.tsx (hunk 3 above)
Comment (tleraj) [reply packages/website/src/common/Chip/index.tsx:112]: It has now become a required property.

File: packages/website/src/constants/sondeConfig.ts (hunk 5 above)
Comment (tleraj) [reply packages/website/src/constants/sondeConfig.ts]: It was `fnu` at the design, but I changed it to `FNU` now :wink:

Comment (ericboucher) [reply packages/website/src/constants/sondeConfig.ts]: Thanks, yeah the designs were not 100% accurate, sorry
